clients/venice-push-job/src/main/java/com/linkedin/venice/hadoop/VenicePushJob.java

```diff
@@ -2307,12 +2311,26 @@ void validateValueSchema(
         .info("Validating value schema: {} for store: {}", pushJobSchemaInfo.getValueSchemaString(), setting.storeName);
 
     SchemaResponse getValueSchemaIdResponse;
-
     if (setting.enableWriteCompute) {
-      getValueSchemaIdResponse = ControllerClient.retryableRequest(
-          controllerClient,
-          setting.controllerRetries,
-          c -> c.getValueOrDerivedSchemaId(setting.storeName, pushJobSchemaInfo.getValueSchemaString()));
+      if (!isUpdateSchema(pushJobSchemaInfo.getValueSchemaString())) {
+        // With new input format, we will need to use the latest update schema to generate partial update record.
+        getValueSchemaIdResponse = ControllerClient.retryableRequest(
+            controllerClient,
+            setting.controllerRetries,
+            c -> c.getLatestUpdateSchema(setting.storeName));
+        // Update the schema to be the update schema of the superset schema.
+        LOGGER.info(
```

Comment:
Shall we validate the input schema align with the latest update schema?
For example, we need to make sure the following:
1. The fields present in the value schema must exist in the latest update schema.
2. The field schema in the value schema must exactly match the corresponding field in the latest update schema.
If the above conditions don't meet, the push job should fail, otherwise the partial update record passing to the backend can be wrong.

Also, let us try to put these validation in some other class, such as `AvroSchemaUtils` to reduce the size of `VenicePushJob`.

Reply:
I'd like to double check about this: For now I added getLatestUpdateSchema endpoint which retrieves partial update schema of the superset schema. We don't have the endpoint to get superset schema yet and I don't plan to add it as this seems a bit duplicated in functionality.

Since update schema contains set field op for each field, I plan to iterate the field in the input (subset) value schema and see if each field type is in update schema and name matches. For union field, we will check if each sub type matches. I think that should work and can be added in to the util class you mentioned, but let me know if you think otherwise.

Reply:
I think we should fetch the latest superset schema and the corresponding update schema, and the validation should happen against the value schema instead of update schema since iterating the fields in update schema will result in pulling wc schema generation related logic to Venice Push Job, which is bad.
```
getAllValueAndDerivedSchema
```
can return all the value schemas, right? I assume it contains the latest superset schema and the corresponding update schema.
Can you leverage this endpoint instead?

Reply:
It is not inside VPJ logic though, but I do agree, validating value field in update schema is a bit weird, even though I can do it in util class.
This endpoint alone does not tell me which schema version is the superset schema. An alternative is use this endpoint and get_store_in_cluster endpoint (and get the superset schema id from store object), so this can get the value schema and corresponding update schema without adding the new get latest update schema I added in this PR.

Reply:
Updated the code and remove the new Controller API